src/test/java/stashpullrequestbuilder/stashpullrequestbuilder/StashRepositoryTest.java

```diff
@@ -437,4 +439,22 @@ public void startJobSkipsNullParameters() {
 
     assertThat(parameters, is(empty()));
   }
+
+  @Test
+  public void constructor_initializes_StashApiClient() {
+    when(trigger.getStashHost()).thenReturn("http://0.0.0.0/");
+    when(trigger.getUsername()).thenReturn("User");
+    when(trigger.getPassword()).thenReturn("****");
+    when(trigger.getProjectCode()).thenReturn("PROJ");
+    when(trigger.getRepositoryName()).thenReturn("Repository");
+    when(trigger.isIgnoreSsl()).thenReturn(false);
+
+    StashRepository newStashRepository = new StashRepository(project, trigger);
+    try {
+      newStashRepository.getTargetPullRequests();
+      fail();
+    } catch (RuntimeException e) {
```

Comment:
This is a bit awkward way to ensure object is setup correctly :) 
Do you think we could somehow test this via getting a positive result? I guess one way would be to use a mock server, like I suggested for #93

Reply:
We can use the factory pattern for StashApiClient. That would solve many issues with the unit tests. We won't need a special `StashRepository` constructor for unit tests. I really consider this unit test a placeholder for a proper unit test.

I can use wiremock in the meantime. Using wiremock would make it an integration test, but it would be less fragile than analyzing a chain of exceptions.

Reply:
Wiremock was the server I had in mind when I wrote "use a mock server" :)

Reply:
Done. Using WireMock now.

I made sure that the unit test fails without the changes to StashRepository.